### ISSUE TITLE
[ENH] Broadcasting of global forecasters

### ISSUE THREAD

**Issue by Xinyu-Wu-0000** [sktime/sktime#6806]
To make the api consistent, we should add broadcasting option in pytorch-forecasting interface. 

And more importantly, we need to decide whether we set `broadcasting` to `True` or `False` by default.

As suggested by @benHeid, we could do it using deprecation cycle to change the default behaviour.

If we are going to set `broadcasting` to `True` by default, we could change the default behaviour of pytorch-forecasting interface.

If we are going to set `broadcasting` to `False` by default, we could change the default behaviour of other models being extended to support global forecasting like huggingface interface and neuralforecast interface.

@benHeid @yarnabrina @fkiraly Any thoughts on this?

**Comment by Xinyu-Wu-0000** [comment]
I would propose to set broadcasting to false by default as most users would expect it when they are using a global forecaster especially a deep learning based one. And this is also the default behavior of upstream repositories such as pytorch-forecasting, huggingface transformers and nueralforecast.

**Comment by yarnabrina** [comment]
I agree that `True` seems to be a more intuitive common default for deep learning forecasters. It has to go through deprecation cycle for `neuralforecast` though.

Opinion: I am not sure if `pytorch-forecasting` or global forecasting was marked as experimental. If it was done, probably deprecation is not necessary, but still can be done.

**Comment by fkiraly** [comment]
Can you kindly explain the usage of the `broadcasting` argument, with examples, please? If there is example code, links are also appreciated.

**Comment by Xinyu-Wu-0000** [comment]
> Can you kindly explain the usage of the `broadcasting` argument, with examples, please? If there is example code, links are also appreciated.

Explaination and examples can be found in [ENH] untangling terminology for "global" forecasting #6690

Explaination: https://github.com/sktime/sktime/issues/6690#issuecomment-2209907759
Example: https://github.com/sktime/sktime/issues/6690#issuecomment-2211606480

**Comment by Xinyu-Wu-0000** [comment]
@benHeid @fkiraly @yarnabrina Shall I add the deprecation messages to #6666 for neuralforecast models and #6686 for huggingface models?

**Comment by yarnabrina** [comment]
It makes sense to me, and I'll agree for `neuralforecast`.

(I haven't used hugging face one myself and didn't follow that development much, so won't comment.)
